### PR TITLE
interface_ospf: interface lookup performance fixes

### PR DESCRIPTION
**Pull request by chrisvanheuveln**
**Summary**
Part 2 of the interface lookup performance work.
Ref #630

- Modified class method InterfaceOspf.interfaces to allow queries for single interfaces.
- interface_ospf.yaml is one of the early implementations and it was still using positional fields instead of named fields. This created a larger effort to implement the interface performance changes.
- interface_ospf.rb is the only consumer of the InterfaceOspf class.

### CODE REVIEW

File: lib/cisco_node_utils/cmd_ref/interface_ospf.yaml [hunk 1]
```diff
@@ -63,7 +63,7 @@ message_digest_key_id:
   default_value: 0
 
 message_digest_key_set:
-  set_value: "%s ip ospf message-digest-key %s %s %s %s"
+  set_value: '<state> ip ospf message-digest-key <keyid> <algtype> <enctype> <enc>'
```
Comment (mikewiebe) [inline lib/cisco_node_utils/cmd_ref/interface_ospf.yaml]: whats the diff between `enctype` and `enc`?

Comment (chrisvanheuveln) [reply lib/cisco_node_utils/cmd_ref/interface_ospf.yaml]: `enctype` == encoding type
`enc` == encoding (aka password)

I have no idea why the original coder used that var name. 'password' would have been more descriptive.

Comment (mikewiebe) [reply lib/cisco_node_utils/cmd_ref/interface_ospf.yaml]: Should we change it to `password`?  Is it more involved then  a search/replace?

Comment (chrisvanheuveln) [reply lib/cisco_node_utils/cmd_ref/interface_ospf.yaml]: oooooh all right. 
Could also change it to `secret_code_word` or `magic_fancy_word` if that flows better.

**Review by mikewiebe — APPROVED** [review]

**Pull request merged**